libraries/hatch-server-user-management/src/AWSCognitoClient.ts

```diff
@@ -1,21 +1,45 @@
-import UserServiceClient, {UserInfo} from './UserServiceClient';
-import {
-  AWS_ACCESS_KEY_ID,
-  AWS_CLIENT_ID,
-  AWS_USER_POOL_ID,
-  AWS_REGION,
-  AWS_SECRET_ACCESS_KEY
-} from './constants';
-import {config, CognitoIdentityServiceProvider} from 'aws-sdk';
+import {AWS_ACCESS_KEY_ID, AWS_CLIENT_ID, AWS_REGION, AWS_SECRET_ACCESS_KEY, AWS_USER_POOL_ID} from './constants';
+import {CognitoIdentityServiceProvider, config} from 'aws-sdk';
 import fetch from 'cross-fetch';
 import jwt from 'jsonwebtoken';
 import jwkToPem from 'jwk-to-pem';
-import {inject, Logger} from '@launchtray/hatch-util';
-import {injectable} from '@launchtray/hatch-util/dist';
+import {inject, injectable, Logger} from '@launchtray/hatch-util';
 import {AttributeListType} from 'aws-sdk/clients/cognitoidentityserviceprovider';
+import {UserManagementClient, UserInfo, UserManagementError, UserManagementErrorCodes} from '@launchtray/hatch-user-management-client';
+import {AWSError} from 'aws-sdk';
+
+const convertAWSErrorToUserManagementError = (awsError: AWSError) => {
+  const message = awsError.code + ' - ' + awsError.message;
+  switch (awsError.code) {
+    case 'CodeMismatchException':
+      return new UserManagementError(UserManagementErrorCodes.INVALID_CODE, message);
+    case 'ExpiredCodeException':
+      return new UserManagementError(UserManagementErrorCodes.INVALID_CODE, message);
+    case 'InvalidPasswordException':
+      return new UserManagementError(UserManagementErrorCodes.INVALID_PASSWORD_FORMAT, message);
+    case 'NotAuthorizedException':
+      if (awsError.message.includes('Password attempts exceeded')) {
+        return new UserManagementError(UserManagementErrorCodes.ACCOUNT_LOCKED, message);
+      } else {
+        return new UserManagementError(UserManagementErrorCodes.UNAUTHORIZED, message);
+      }
+    case 'PasswordResetRequiredException':
+      return new UserManagementError(UserManagementErrorCodes.ACCOUNT_LOCKED, message);
+    case 'TokenExpiredException':
+      return new UserManagementError(UserManagementErrorCodes.EXPIRED_TOKEN, message);
+    case 'UsernameExistsException':
+      return new UserManagementError(UserManagementErrorCodes.USERNAME_EXISTS, message);
+    case 'UserNotConfirmedException':
+      return new UserManagementError(UserManagementErrorCodes.USER_NOT_CONFIRMED, message);
+    case 'UserNotFoundException':
```

Comment:
Weren't we going to map this to UNAUTHORIZED?

Reply:
Per offline discussion, we will leave as is for unauthenticated API calls